### DISCUSSION TITLE
Find transcript on Deepgram site

### DISCUSSION THREAD

**Discussion by bhaviland** [deepgram/community#130] [General help] [answered]
I don't know anything about coding (have forgotten Fortran :) ). I want to use Deepgram to simply create transcriptions from an online video course (have permission). I created the transcription but had to take time away and got logged off. When I went back I can see that the transcription was done as per below, but I can't find where to download it as a document. Can anyone help? Have not been able to get an answer from support.

![image](https://user-images.githubusercontent.com/90063713/234078916-b0f80928-a758-4f1b-9648-b8e635a49d97.png)

**Comment by briancbarrow** [comment, marked as answer]
Hi @bhaviland,

Deepgram does not store transcriptions. You will need to have your code handle the response or provide a callback url to handle the response.

**Reply by bhaviland** [reply]
Wow! Thanks for letting me know. I guess it's really a site for programmers which I am not so I wouldn't know how to do what you suggest. I was able to copy previous transcripts, but I was there when they finished which I was not this time. I'll have to set an alarm for myself. Appreciate the info.

**Reply by scottstephenson** [reply]
Learning to code again can be daunting especially if you've been away from it for a while. But it often isn’t as hard as it seems. Python is a great language to start with. It's simple and readable, yet very powerful. The language is designed to be easy to understand and write, so you might find that it's easier to pick up than you expect or remember from the FORTRAN days.

Below, I have provided two Python scripts: 

1. The first script is to upload a local audio file to Deepgram and save the transcript as a text file in the same location as the original audio file.
2. The second script is to generate an SRT file for closed captioning on a web video from the Deepgram transcript.

**1. Upload a local audio file and save the transcript:**

```python
import requests
import json
import os

# Replace with your Deepgram API Key
api_key = "YOUR_DEEPGRAM_API_KEY"

# Endpoint for Deepgram's transcription service
url = "https://api.deepgram.com/v1/listen"

# Headers for the request
headers = {
    "Authorization": "Token " + api_key
}

# Path to your local audio file
audio_file_path = "/path/to/your/audio/file.wav"

# Open the audio file in read-bytes mode
with open(audio_file_path, "rb") as audio_file:
    # Send the request to Deepgram
    response = requests.post(url, headers=headers, data=audio_file)

# Parse the response from Deepgram
data = json.loads(response.text)

# Extract the transcript text
transcript_text = data["results"]["channels"][0]["alternatives"][0]["transcript"]

# Determine the path for the output text file
output_file_path = os.path.splitext(audio_file_path)[0] + '.txt'

# Write the transcript text to the output file
with open(output_file_path, 'w') as output_file:
    output_file.write(transcript_text)

print(f'Transcript saved to: {output_file_path}')
```

**2. Generate an SRT file for closed captioning from the Deepgram transcript:**

```python
import requests
import os
import json

def transcribe_audio(file_path):
    # Define the Deepgram API endpoint
    url = "https://api.deepgram.com/v1/listen"

    # Replace YOUR_DEEPGRAM_API_KEY with your actual Deepgram API key
    headers = {"Authorization": "Token YOUR_DEEPGRAM_API_KEY"}

    # Open the audio file and send a POST request to Deepgram
    with open(file_path, 'rb') as audio_file:
        response = requests.post(url, headers=headers, data=audio_file)
        
    # If the request failed, this line will raise an error
    response.raise_for_status()

    # Parse the JSON response and return it
    return response.json()

def generate_srt(transcript, srt_file_path):
    # Open the output file in write mode
    with open(srt_file_path, 'w') as srt_file:
        i = 1  # This variable will keep track of the subtitle index

        # For each channel in the transcript...
        for channel in transcript['results']['channels']:
            # ...and for each alternative in the channel...
            for alternative in channel['alternatives']:
                words = alternative['words']  # The list of words in this alternative
                
                # While there are words left in the alternative...
                while words:
                    # Start a new line with the first word
                    start_time = words[0]['start']
                    end_time = words[0]['end']
                    line = words[0]['word']

                    # Remove the first word from the list (which is now the last one that was added to the line)
                    words.pop(0)

                    # Convert the start and end times to hours, minutes, and seconds
                    start_sec, start_min = divmod(start_time, 60)
                    start_hours, start_min = divmod(start_min, 60)
                    end_sec, end_min = divmod(end_time, 60)
                    end_hours, end_min = divmod(end_min, 60)

                    # Format the subtitle and write it to the file
                    caption = "{}\n{:02}:{:02}:{:02},000 --> {:02}:{:02}:{:02},000\n{}\n\n".format(
                        i, int(start_hours), int(start_min), int(start_sec), int(end_hours), int(end_min), int(end_sec), line)
                    srt_file.write(caption)

                    i += 1  # Increment the subtitle index

# Replace this with the path to your audio file
audio_file_path = "/path/to/your/audio/file.wav"

# The SRT file will have the same name and location as the audio file, but with .srt extension
srt_file_path = os.path.splitext(audio_file_path)[0] + ".srt"

# Transcribe the audio file and generate the SRT file
transcript = transcribe_audio(audio_file_path)
generate_srt(transcript, srt_file_path)

# Save the transcript as a JSON file
with open(os.path.splitext(audio_file_path)[0] + ".json", 'w') as json_file:
    json.dump(transcript, json_file, indent=4)

# Print the path to the SRT file
print(f"SRT file saved as: {srt_file_path}")

# Print the path to the JSON file
print(f"JSON file saved as: {os.path.splitext(audio_file_path)[0] + '.json'}")
```

Please replace `"YOUR_DEEPGRAM_API_KEY"` with your actual Deepgram API key and `/path/to/your/audio/file.wav` with the actual path of the audio file you want to transcribe on your local machine.

You can run these scripts in any Python environment. If you don't have one installed, you might want to try downloading [Anaconda Individual Edition](https://www.anaconda.com/products/individual). It's free and comes with a lot of useful tools for Python programming.

Once you've installed a Python environment, you can save these scripts as `.py` files (for example, `transcribe_audio.py` and `generate_srt.py`), and then run them from the command line by navigating to the directory where you saved them and typing `python transcribe_audio.py` or `python generate_srt.py`. Alternatively, you can run them directly in your Python environment's script editor.

I hope this helps!

**Comment by bhaviland** [comment]
Hi Scott,

Appreciate your response and the coding tips. I kept running the transcript and for some reason the last time I was able to use the copy option to get the document. So I'm OK for now but I will save your info in case I need to do this in the future.

Cheers,
Brian
________________________________
